### PR TITLE
fix isPaymentDone verification - verify by recovered address not reci…

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

**Comment by AndriianChestnykh** [comment]
@volodymyr-basiuk Additionally, please add NatSpec comments everywhere in the new payment functionality and check if verification will be run in the deploy scripts at deployment time in the same way as it is run for other contracts

File: contracts/payment/MCPayment.sol [hunk 1]
```diff
@@ -112,8 +112,8 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
         Iden3PaymentRailsRequestV1 memory paymentData,
         bytes memory signature
     ) external payable {
-        verifyIden3PaymentRailsRequestV1Signature(paymentData, signature);
-        bytes32 paymentId = keccak256(abi.encode(paymentData.recipient, paymentData.nonce));
+        address recoverd = recoverIden3PaymentRailsRequestV1Signature(paymentData, signature);
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply contracts/payment/MCPayment.sol]: signer is the better name, changed

File: contracts/payment/MCPayment.sol [hunk 2]
```diff
@@ -187,18 +187,18 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
             r,
             s
         );
-        _transferERC20(paymentData);
+        _transferERC20(paymentData, recovered);
     }
 
-    function isPaymentDone(address recipient, uint256 nonce) external view returns (bool) {
+    function isPaymentDone(address signer, uint256 nonce) external view returns (bool) {
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply contracts/payment/MCPayment.sol]: issuer now always sign the message, so we can use `issuer`

File: test/payment/mc-payment.test.ts [hunk 3]
```diff
@@ -59,11 +60,12 @@ describe("MC Payment Contract", () => {
     const signature = await issuer1Signer.signTypedData(domainData, types, paymentData);
     const verifyGas = await payment
       .connect(userSigner)
-      .verifyIden3PaymentRailsRequestV1Signature.estimateGas(paymentData, signature);
+      .recoverIden3PaymentRailsRequestV1Signature.estimateGas(paymentData, signature);
     console.log("Verification Gas: " + verifyGas);
```
[the remark this answers is not in the document]
Comment (volodymyr-basiuk) [reply test/payment/mc-payment.test.ts]: not really, removed

**Review by vmidyllic — APPROVED** [review]

**Review by AndriianChestnykh — APPROVED** [review]

**Pull request merged**